crmd/remote_lrmd_ra.c

```diff
@@ -536,9 +536,40 @@ remote_lrm_op_callback(lrmd_event_data_t * op)
         (ra_data->cur_cmd == NULL) &&
         (ra_data->active == TRUE)) {
 
-        crm_err("Unexpected disconnect on remote-node %s", lrm_state->node_name);
-        ra_data->recurring_cmds = fail_all_monitor_cmds(ra_data->recurring_cmds);
-        ra_data->cmds = fail_all_monitor_cmds(ra_data->cmds);
+        if (crmd_is_rsc_managed(lrm_state->node_name)) {
+            crm_err("Unexpected disconnect on remote-node %s", lrm_state->node_name);
+            ra_data->recurring_cmds = fail_all_monitor_cmds(ra_data->recurring_cmds);
+            ra_data->cmds = fail_all_monitor_cmds(ra_data->cmds);
+        } else {
+            crm_info("Disconnect on unmanaged remote-node %s", lrm_state->node_name);
+            /* Do roughly what a 'stop' on the remote-resource would do */
+            g_hash_table_remove_all(lrm_state->pending_ops);
+            ra_data->active = FALSE;
+            lrm_state_disconnect(lrm_state);
+            if (ra_data->cmds) {
+                g_list_free_full(ra_data->cmds, free_cmd);
+            }
+            if (ra_data->recurring_cmds) {
+                g_list_free_full(ra_data->recurring_cmds, free_cmd);
+            }
+            ra_data->cmds = NULL;
+            ra_data->recurring_cmds = NULL;
+            remote_node_down(lrm_state->node_name, DOWN_KEEP_LRM);
+            {
+                lrmd_event_data_t op = { 0, };
+
+                /* now fake the reply of a successful 'stop' */
+                op.type = lrmd_event_exec_complete;
+                op.rsc_id = lrm_state->node_name;
+                op.op_type = "stop";
+                op.rc = PCMK_OCF_OK;
+                op.op_status = PCMK_LRM_OP_DONE;
+                op.t_run = time(NULL);
+                op.t_rcchange = op.t_run;
+                op.call_id = generate_callid();
+                lrm_op_callback(&op);
+            }
+        }
```

Comment:
oops had to rework that a little. With lrm_op_callback if would search search for the local
lrm_state (what we want to have as the remote-resource should stop locally of course)
and using process_lrm_event with the remote lrm_state will of course not have the
desired effect. Changed the interface so that you can give the lrm_state if you already
have the proper one (which we don't have here) and otherwise it will get the local
lrm_state. Not having the remote lrm_state now anymore and to make it more
versatile for maybe future use the rsc_id to succed is given now as well.